src/Workspaces/Core/Desktop/Workspace/MSBuild/MSBuildProjectLoader.cs

```diff
@@ -259,8 +259,12 @@ private class LoadState
             private Dictionary<ProjectId, ProjectInfo> _projectIdToProjectInfoMap
                 = new Dictionary<ProjectId, ProjectInfo>();
 
-            private List<ProjectInfo> _projectInfoList
-                = new List<ProjectInfo>();
+            /// <summary>
+            /// Used to memoize results of <see cref="ProjectAlreadyReferencesProject"/> calls.
+            /// Reset any time internal state is changed.
+            /// </summary>
+            private Dictionary<ProjectId, Dictionary<ProjectId, bool>> ProjectAlreadyReferencesProjectMemo
```

Comment:
The name suggestion was also based on my expectation that it could be solve more simply.  Maybe instead, _projectAlreadyReferencesProjectResultCache.

---
In reply to: [81627757](https://github.com/dotnet/roslyn/pull/14245#discussion_r81627757) [](ancestors = 81627757)